go/vt/sqlparser/sql.y

```diff
@@ -8489,6 +8505,13 @@ ignore_opt:
 | IGNORE
   { $$ = true }
 
+ignore_or_replace_opt:
+  { $$ = "" }
+| IGNORE
+  { $$ = "ignore" }
+| REPLACE
+  { $$ = "replace" }
+
```

Comment:
I think we should introduce an enum style integer for this instead of using strings. See for example this one:

```
type PartitionByType int8

// Constant for Enum Type - PartitionByType
const (
	HashType PartitionByType = iota
	KeyType
	RangeType
	ListType
)
```